announcement/announcement-impl/impl/src/java/org/sakaiproject/announcement/impl/SiteEmailNotificationAnnc.java

```diff
@@ -423,38 +423,45 @@ public void execute(String opaqueContext)
 	{
 		// get the message
 		final Reference ref = entityManager.newReference(opaqueContext);
-		
-		// needed to access the message
-		enableSecurityAdvisorToGetAnnouncement();
-		
-		final AnnouncementMessage msg = (AnnouncementMessage) ref.getEntity();
-		final AnnouncementMessageHeader hdr = (AnnouncementMessageHeader) msg.getAnnouncementHeader();
-
-		// read the notification options
-		final String notification = msg.getProperties().getProperty("notificationLevel");
-
-		int noti = NotificationService.NOTI_OPTIONAL;
-		if ("r".equals(notification))
-		{
-			noti = NotificationService.NOTI_REQUIRED;
-		}
-		else if ("n".equals(notification))
-		{
-			noti = NotificationService.NOTI_NONE;
-		}
+		try {
+			// needed to access the message
+			enableSecurityAdvisorToGetAnnouncement();
 			
-		final Event delayedNotificationEvent = eventTrackingService.newEvent("annc.schInv.notify", msg.getReference(), true, noti);
-//		eventTrackingService.post(event);
-
-		NotificationEdit notify = notificationService.addTransientNotification();
-		
-		super.notify(notify, delayedNotificationEvent);
-
-		// since we build the notification by accessing the
-		// message within the super class, can't remove the
-		// SecurityAdvisor until this point
-		// done with access, need to remove from stack
-		disableSecurityAdvisor();
+			final AnnouncementMessage msg = (AnnouncementMessage) ref.getEntity();
+			final AnnouncementMessageHeader hdr = (AnnouncementMessageHeader) msg.getAnnouncementHeader();
+	
+			// read the notification options
+			final String notification = msg.getProperties().getProperty("notificationLevel");
+	
+			int noti = NotificationService.NOTI_OPTIONAL;
+			if ("r".equals(notification))
+			{
+				noti = NotificationService.NOTI_REQUIRED;
+			}
+			else if ("n".equals(notification))
+			{
+				noti = NotificationService.NOTI_NONE;
+			}
+				
+			final Event delayedNotificationEvent = eventTrackingService.newEvent("annc.schInv.notify", msg.getReference(), true, noti);
+			//eventTrackingService.post(event);
+	
+			NotificationEdit notify = notificationService.addTransientNotification();
+			
+			super.notify(notify, delayedNotificationEvent);
+			
+		} catch (NullPointerException npe) {
```

Comment:
Personally I think it's much better todo the explicit null check where you expect it, rather than surrounding the whole block of code and catching a NPE, in this case checking to see if `msg` is null.